message_definitions/v1.0/ardupilotmega.xml

```diff
@@ -210,6 +210,16 @@
         <param index="6">Magic number</param>
         <param index="7">Magic number</param>
       </entry>
+      <entry value="42600" name="MAV_CMD_DO_WINCH">
+        <description>Command to operate winch</description>
+        <param index="1">winch number (0 for the default winch, otherwise a number from 1 to max number of winches on the vehicle)</param>
+        <param index="2">action (0=disable, 1=length control, 2=rate control.  See WINCH_ACTIONS enum)</param>
```

Comment:
I'd remove the 0=disable, 1=length control etc. It's not scalable, and it's why we have the comment that tells you to lookup the enum.

Reply:
Actually I take it back. I find this to be a confusing command. Why do you need the action at all? You already have full control over both length and rate in the params. If you only want to control the one why not just set the other to 0? (or NaN for no control)

Reply:
"0" has a real meaning for both rate-control and length-control so I don't think we can use that to mean "ignore".  For example maintaining the rope at it's current length is action=length-control with length of 0.  I'm definitely not a fan of asking users to enter NaN.  I suspect many users don't know what NaN is.
I agree that we could use length-control and then use the rate field to specific the desired rate at which we get to the desired length.  I haven't implemented that for now but it could be added later.

Reply:
Length control is purely a relative length command as you've implemented it here. The proposed XML file doesn't allow the usage of both rate and length control as presented. (And if it does then why is param2 used for a enum?

Reply:
i've added the length+rate control now.
one way or the other, we need to have an action field to specify relaxing the winch.